### PR TITLE
[risk=low][no ticket] Migrate Leo AppsApi calls to new client

### PR DESCRIPTION
**Pull request by jmthibault79**
Tested these calls from local UI to confirm that they function as expected:
* createApp
* listAppsInWorkspace
* deleteApp
* cron checkPersistentDisks (listAppsInProjectAsService)
* WorkspaceAdminController adminListUserAppsInWorkspace (listAppsInProjectAsService)

Removed because unused:
* getApp

TODO - I can't seem to trigger egress locally
* EgressRemediationService stopUserRuntimesAndApps (deleteUserAppsAsService)

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 1]
```diff
@@ -159,36 +156,43 @@ ListRuntimeResponse toApiListRuntimeResponse(
...
-      source = "cloudContext",
-      qualifiedByName = "legacy_cloudContextToGoogleProject")
   Runtime toApiRuntime(LeonardoListRuntimeResponse runtime);
 
   RuntimeError toApiRuntimeError(LeonardoClusterError err);
 
   @AfterMapping
   default void getRuntimeAfterMapper(
       @MappingTarget Runtime runtime, LeonardoGetRuntimeResponse leonardoGetRuntimeResponse) {
-    mapRuntimeLabels(runtime, leonardoGetRuntimeResponse.getLabels());
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:191]: made this call more explicit - see the references to mapRuntimeConfigurationLabels

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 2]
```diff
@@ -212,75 +215,65 @@ default void listRuntimeAfterMapper(
   @Mapping(
       target = "googleProject",
       source = "cloudContext",
-      qualifiedByName = "legacy_cloudContextToGoogleProject")
+      qualifiedByName = "cloudContextToGoogleProject")
+  @Mapping(target = "appType", source = "labels", qualifiedByName = "mapAppType")
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:219]: removed the indirection to `getAppAfterMapper()`

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 3]
```diff
@@ -212,75 +215,65 @@ default void listRuntimeAfterMapper(
...
   @Mapping(target = "creator", source = "auditInfo.creator")
-  @Mapping(target = "autopauseThreshold", ignore = true)
   @Mapping(
       target = "googleProject",
       source = "cloudContext",
-      qualifiedByName = "legacy_cloudContextToGoogleProject")
-  @Mapping(target = "appType", ignore = true)
-  UserAppEnvironment toApiApp(LeonardoListAppResponse app);
+      qualifiedByName = "cloudContextToGoogleProject")
+  @Mapping(target = "appType", source = "labels", qualifiedByName = "mapAppType")
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:230]: removed the indirection to `listAppsAfterMapper()`

File: api/src/main/java/org/pmiops/workbench/api/AppsController.java [hunk 4]
```diff
@@ -66,14 +66,6 @@ public ResponseEntity<EmptyResponse> deleteApp(
     return ResponseEntity.ok(new EmptyResponse());
   }
 
-  @Override
-  public ResponseEntity<UserAppEnvironment> getApp(String workspaceNamespace, String appName) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/AppsController.java:70]: this is unused.  The UI never asks for an app by name.  Instead we always list all apps in a workspace.

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 5]
```diff
@@ -679,15 +676,6 @@ public void createApp(CreateAppRequest createAppRequest, DbWorkspace dbWorkspace
         });
   }
 
-  @Override
-  public UserAppEnvironment getAppByNameByProjectId(String googleProjectId, String appName) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:683]: unused

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 6]
```diff
@@ -159,36 +155,43 @@ ListRuntimeResponse toApiListRuntimeResponse(
...
   @Mapping(target = "toolDockerImage", source = "runtimeImages")
-  @Mapping(target = "configurationType", ignore = true)
-  @Mapping(target = "gceConfig", ignore = true)
-  @Mapping(target = "gceWithPdConfig", ignore = true)
-  @Mapping(target = "dataprocConfig", ignore = true)
   @Mapping(
       target = "googleProject",
       source = "cloudContext",
       qualifiedByName = "legacy_cloudContextToGoogleProject")
+  @Mapping(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:162]: unrelated cleanup: I was able to make this mapping more explicit

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 7]
```diff
@@ -208,79 +210,57 @@ default void listRuntimeAfterMapper(
...
-  @Mapping(target = "appName", source = "appName")
   @Mapping(
       target = "googleProject",
       source = "cloudContext",
-      qualifiedByName = "legacy_cloudContextToGoogleProject")
+      qualifiedByName = "cloudContextToGoogleProject")
+  @Mapping(target = "appType", source = "labels", qualifiedByName = "mapAppType")
   @Mapping(target = "autopauseThreshold", ignore = true)
-  @Mapping(target = "appType", ignore = true)
-  UserAppEnvironment toApiApp(LeonardoGetAppResponse app);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:218]: unused after removing getApp(), so removed

File: api/src/test/java/org/pmiops/workbench/leonardo/LeonardoApiClientTest.java [hunk 8]
```diff
@@ -358,12 +367,6 @@ public void testCreateAppFail_newPd_pdAlreadyExist() throws Exception {
                 createAppRequest.persistentDiskRequest(persistentDiskRequest), testWorkspace));
   }
 
-  @Test
-  public void testGetAppSuccess() throws Exception {
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/leonardo/LeonardoApiClientTest.java:362]: no longer used

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java [hunk 9]
```diff
@@ -208,79 +210,57 @@ default void listRuntimeAfterMapper(
...
-
-  default void setAppType(UserAppEnvironment appEnvironment, @Nullable Object appLabels) {
-    appEnvironment.setAppType(
-        LeonardoLabelHelper.maybeMapLeonardoLabelsToGkeApp(appLabels)
-            .orElseThrow(
-                () ->
-                    new ServerErrorException(
-                        String.format(
-                            "Missing app type labels for app with labels %s", appLabels))));
+  @Named("mapAppType")
```
Comment (evrii) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:242]: I don't see this map being used before. What is this for?

**Review by evrii — APPROVED** [review]
Looks good to me. I tested this locally by starting an app, listing apps(app side panel), listing apps on admin page, and deleting an app.

File: api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java (hunk 9 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/utils/mappers/LeonardoMapper.java:242]: This was part of `listAppsAfterMapper()` which was implicitly called as an `@AfterMapping` after the main `toApiApp(LeonardoListAppResponse app)` was called.

AfterMappings are a bit error-prone to work with, so it's better to make the association explicit where possible.  This new `mapAppType()` is that more explicit version.
